src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -163,6 +163,21 @@ private PatternSyntax ParseNegatedPattern(Precedence precedence, bool afterIs, b
                     ConvertToKeyword(this.EatToken()),
                     ParseNegatedPattern(precedence, afterIs, inSwitchArmPattern));
             }
+            else if (this.CurrentToken.Kind == SyntaxKind.EqualsEqualsToken)
+            {
+                return AddLeadingSkippedSyntax(
+                    skippedSyntax: AddError(this.EatToken(), ErrorCode.ERR_EqualityOperatorInPatternNotSupported),
+                    node: ParsePrimaryPattern(precedence, afterIs, inSwitchArmPattern));
```

Comment:
When we parse a `not` pattern, we allow nested `not`; should we allow that here too?

Reply:
Done.